### PR TITLE
add SeccompProfile to Pod and Container accessors/mutators

### PR DESCRIPTION
**Pull request by stlaz**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR adds the `SeccompProfile` field into the pod and container security context accessor/mutator to maintain the securityContext workflow.

#### Which issue(s) this PR fixes:

Related to https://github.com/kubernetes/kubernetes/issues/91286

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/cc @saschagrunert @tallclair
since you seem to have been active at the referenced issue

### CODE REVIEW

File: pkg/securitycontext/accessors.go [hunk 1]
```diff
@@ -171,6 +173,19 @@ func (w *podSecurityContextWrapper) SetRunAsNonRoot(v *bool) {
...
 	w.podSC.RunAsNonRoot = v
 }
+func (w *podSecurityContextWrapper) SeccompProfile() *api.SeccompProfile {
+	if w.podSC == nil {
+		return nil
+	}
+	return w.podSC.SeccompProfile
+}
+func (w *podSecurityContextWrapper) SetSeccompProfile(p *api.SeccompProfile) {
+	if w.podSC == nil && p == nil {
```
Comment (saschagrunert) [inline pkg/securitycontext/accessors.go:183]: Shouldn't this be logical `or`?
```suggestion
	if w.podSC == nil || p == nil {
```

**Review by saschagrunert — COMMENTED** [review]
/ok-to-test
/sig node

File: pkg/securitycontext/accessors.go (hunk 1 above)
Comment (stlaz) [reply pkg/securitycontext/accessors.go:183]: You would end up not setting the profile even if `p` was actually set.

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**